### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by TotallyGamerJet**
This PR contains quite a few quality of life cleanups. This is in anticipation of the objc PR

- remove unused forward declarations
- check if fn is nil in SyscallN
- rename runtime.go to go_runtime.go
- use go offsets in assembly
- avoid copying null terminated string
- runtime_libcCall now takes fn as uintptr
- format assembly with asmfmt
- make README.md useful
- separate dl files into os and non-os dependent parts
- remove call to non-portable libc_error (__error)

### CODE REVIEW

**Comment by TotallyGamerJet** [comment]
Should the license header be changed to Ebitengine?
`// SPDX-FileCopyrightText: 2022 The Ebiten Authors` >> `// SPDX-FileCopyrightText: 2022 The Ebitengine Authors`

File: README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# purego
+[![Go Reference](https://pkg.go.dev/badge/github.com/ebitengine/purego?GOOS=darwin.svg)](https://pkg.go.dev/github.com/ebitengine/purego?GOOS=darwin)
+
+A library for calling C functions from Go without CGO.
```
Comment (hajimehoshi) [inline README.md:4]: Cgo

File: syscall.go [hunk 2]
```diff
@@ -21,6 +21,9 @@ const maxArgs = 9
 //go:nosplit
 //go:uintptrescapes
 func SyscallN(fn uintptr, args ...uintptr) (r1, r2, err uintptr) {
+	if fn == 0 {
+		panic("fn is nil")
```
Comment (hajimehoshi) [inline syscall.go:25]: `purego: fn is nil`

File: syscall.go [hunk 3]
```diff
@@ -21,6 +21,9 @@ const maxArgs = 9
...
+	}
 	if len(args) > maxArgs {
 		panic("too many arguments to SyscallN")
```
Comment (hajimehoshi) [inline syscall.go:28]: `purego: too many arguments to SyscallN`

**Comment by hajimehoshi** [comment]
> Should the license header be changed to Ebitengine?
> // SPDX-FileCopyrightText: 2022 The Ebiten Authors >> // SPDX-FileCopyrightText: 2022 The Ebitengine Authors

Yes, not mandatory but preferable!

**Review by hajimehoshi — APPROVED** [review]
LGTM, thanks!

**Pull request merged**